mono/metadata/domain.c

```diff
@@ -1217,7 +1217,9 @@ mono_domain_free (MonoDomain *domain, gboolean force)
 
 	domain->setup = NULL;
 
+#ifndef HAVE_BOEHM_GC
```

Comment:
No, the matching `#if` check is same: https://github.com/Unity-Technologies/mono/blob/7f15b539954c6b2f8591ce78877c0d1872b4d93c/mono/metadata/domain.c#L387